webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx

```diff
@@ -0,0 +1,563 @@
+import { placeholderMenuItems, TableActionMenu, TableActionMenuProps } from './menu';
+import { MapOfIdsToColors } from './useGlasbey';
+import { getDurationInEnglish, getTheme, getTimeInEnglish, headerIcons } from './utils';
+
+const GRID_HEIGHT = 700;
+const cells: DataEditorProps['customRenderers'] = [
+  SparklineCell,
+  TagsCell,
+  UserProfileCell,
+  SpinnerCell,
+  RangeCell,
+  LinksCell,
+];
+
+interface Props {
+  colorMap: MapOfIdsToColors;
+  data: ExperimentItem[];
+
+  handleScroll?: (r: Rectangle) => void;
+
+  sortableColumnIds: ExperimentColumn[];
+  setSortableColumnIds: Dispatch<SetStateAction<ExperimentColumn[]>>;
+
+  selectedExperimentIds: string[];
+  setSelectedExperimentIds: Dispatch<SetStateAction<string[]>>;
+  selectAll: boolean;
+  setSelectAll: Dispatch<SetStateAction<boolean>>;
+}
+
+const STATIC_COLUMNS: ExperimentColumn[] = ['selected', 'name'];
+
+type ColumnDef = SizedGridColumn & {
+  id: ExperimentColumn;
+  isNumerical?: boolean;
+  renderer: (record: ExperimentItem, idx: number) => GridCell;
+};
+
+export const GlideTable: React.FC<Props> = ({
+  data,
+  setSelectedExperimentIds,
+  sortableColumnIds,
+  setSortableColumnIds,
+  colorMap,
+  selectAll,
+  setSelectAll,
+  handleScroll,
+}) => {
+  const gridRef = useRef(null);
+
+  const [menuIsOpen, setMenuIsOpen] = useState(false);
+
+  const handleMenuClose = useCallback(() => {
+    setMenuIsOpen(false);
+  }, []);
+
+  const {
+    ui: { darkLight },
+  } = useUI();
+
+  const [menuProps, setMenuProps] = useState<Omit<TableActionMenuProps, 'open'>>({
+    handleClose: handleMenuClose,
+    x: 0,
+    y: 0,
+  });
+
+  const users = Loadable.map(useObservable(usersStore.getUsers()), ({ users }) => users);
+
+  const columnIds = useMemo<ExperimentColumn[]>(
+    () => [...STATIC_COLUMNS, ...sortableColumnIds],
+    [sortableColumnIds],
+  );
+  const navigate = useNavigate();
+  const bodyStyles = getComputedStyle(document.body);
+
+  const [selection, setSelection] = React.useState<GridSelection>({
+    columns: CompactSelection.empty(),
+    rows: CompactSelection.empty(),
+  });
+  const getRowThemeOverride = React.useCallback(
+    (row: number): Partial<Theme> | undefined =>
+      colorMap[data[row]?.id]
+        ? {
+            accentColor: colorMap[data[row]?.id],
+            borderColor: '#F0F0F0',
+          }
+        : { borderColor: '#F0F0F0' },
+    [colorMap, data],
+  );
+
+  useEffect(() => {
+    const selectedRowIndices = selection.rows.toArray();
+    setSelectedExperimentIds((prevIds) => {
+      const selectedIds = selectedRowIndices
+        .map((idx) => String(data?.[idx]?.id))
+        .filter((x) => x !== undefined);
+      if (prevIds === selectedIds) return prevIds;
+      return selectedIds;
+    });
+  }, [selection.rows, setSelectedExperimentIds, data]);
+
+  const theme = getTheme(bodyStyles);
+
+  const [columnWidths, setColumnWidths] = useState<Record<ExperimentColumn, number>>({
```

Comment:
i think width should be winder for some columns. 

<img width="278" alt="image" src="https://user-images.githubusercontent.com/109113616/228912642-b66f619c-2207-4238-8ee9-411791e369ed.png">

Reply:
thanks for reviewing @keita-determined . i am going to roll the style changes into the next PR. so i added your comments here: 

https://hpe-aiatscale.atlassian.net/browse/WEB-1108?focusedCommentId=20935

the squishedness is already part of that ticket